File: java/javafind/src/test/java/javafind/FindFileTest.java
package javafind;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class FindFileTest {

    public FindFileTest() {}

    @Test
    public final void test_findfile_abs_path() {
        String path = "~/src/xfind/java/javafind/src/main/java/javafind";
        String filename = "FindFile.java";
        FindFile findFile = new FindFile(path, filename, FileType.CODE);
        String expectedFilePath = path + "/" + filename;
        assertEquals(expectedFilePath, findFile.toString());
    }

    @Test
    public final void test_findfile_rel_path1() {
        String path = ".";
        String filename = "FindFile.java";
        FindFile findFile = new FindFile(path, filename, FileType.CODE);
        String expectedFilePath = path + "/" + filename;
        assertEquals(expectedFilePath, findFile.toString());
    }

    @Test
    public final void test_findfile_rel_path2() {
        String path = "..";
        String filename = "FindFile.java";
        FindFile findFile = new FindFile(path, filename, FileType.CODE);
        String expectedFilePath = path + "/" + filename;
        assertEquals(expectedFilePath, findFile.toString());
    }
}

File: kotlin/ktfind/src/test/kotlin/ktfind/FindFileTest.kt
package ktfind

import org.junit.Assert.assertEquals
import org.junit.Test
import java.io.File

class FindFileTest {

    @Test
    fun test_findfile_abs_path() {
        val path = "/Users/cary/src/xfind/kotlin/ktfind/src/main/kotlin/ktfind/FindFile.kt"
        val findFile = FindFile(File(path), FileType.CODE)
        assertEquals(path, findFile.toString())
    }

    @Test
    fun test_findfile_tilde_path() {
        val path = "~/src/xfind/kotlin/ktfind/src/main/kotlin/ktfind/FindFile.kt"
        val findFile = FindFile(File(path), FileType.CODE)
        assertEquals(path, findFile.toString())
    }

    @Test
    fun test_findfile_rel_path1() {
        val path = "./FindFile.kt"
        val findFile = FindFile(File(path), FileType.CODE)
        assertEquals(path, findFile.toString())
    }

    @Test
    fun test_findfile_rel_path2() {
        val path = "../FindFile.kt"
        val findFile = FindFile(File(path), FileType.CODE)
        assertEquals(path, findFile.toString())
    }
}
